src/cpuid/src/common.rs

```diff
@@ -91,6 +92,39 @@ pub fn get_vendor_id_from_cpuid(cpuid: &CpuId) -> Result<[u8; 12], Error> {
     Err(Error::NotSupported)
 }
 
+/// Validates that a CPUID and another leaf 0x1 describe the same
+/// CPU in terms of manufacturer and model.
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+pub fn is_same_model(cpuid: &CpuId, other_leaf0x1_eax: u32) -> bool {
```

Comment:
I wouldn't supply the other `0x1` leaf as a parameter. I would read it inside this method.